unix/BUILD.bazel

```diff
@@ -0,0 +1,208 @@
+        "stub_find_word_to_complete.h",
+        "stub_fn_complete.h",
+        "wrap-chared.c",
+        "wrap-chartype.c",
+        "wrap-common.c",
+        "wrap-el.c",
+        "wrap-eln.c",
+        "wrap-emacs.c",
+        "wrap-filecomplete.c",
+        "wrap-hist.c",
+        "wrap-history.c",
+        "wrap-historyn.c",
+        "wrap-keymacro.c",
+        "wrap-map.c",
+        "wrap-parse.c",
+        "wrap-prompt.c",
+        "wrap-read.c",
+        "wrap-readline.c",
+        "wrap-refresh.c",
+        "wrap-search.c",
+        "wrap-sig.c",
+        "wrap-strlcat.c",
+        "wrap-strlcpy.c",
+        "wrap-terminal.c",
+        "wrap-tokenizer.c",
+        "wrap-tokenizern.c",
+        "wrap-tty.c",
+        "wrap-unvis.c",
+        "wrap-vi.c",
+        "wrap-vis.c",
+        "wrap-wcsdup.c",
+    ],
+    cgo = True,
+    clinkopts = select({
+        "@io_bazel_rules_go//go/platform:android": [
+            "-lncurses",
+        ],
+        "@io_bazel_rules_go//go/platform:darwin": [
+            "-ledit",
+        ],
+        "@io_bazel_rules_go//go/platform:dragonfly": [
+            "-ledit",
+        ],
+        "@io_bazel_rules_go//go/platform:freebsd": [
+            "-ledit",
+        ],
+        "@io_bazel_rules_go//go/platform:ios": [
+            "-ledit",
+        ],
+        "@io_bazel_rules_go//go/platform:linux": [
+            "-lncurses",
+        ],
+        "@io_bazel_rules_go//go/platform:netbsd": [
+            "-ledit",
+        ],
+        "@io_bazel_rules_go//go/platform:openbsd": [
+            "-ledit",
+        ],
+        "//conditions:default": [],
+    }),
+    copts = select({
+        "@io_bazel_rules_go//go/platform:android": [
+            "-Iunix/src -Iunix/src/c-libedit -Iunix/src/c-libedit/editline -Iunix/src/c-libedit/linux-build -D_GNU_SOURCE",
+            "-Wno-unused-result -Wno-pointer-sign",
+        ],
+        "@io_bazel_rules_go//go/platform:darwin": [
+            "-Iunix/shim",
+        ],
+        "@io_bazel_rules_go//go/platform:dragonfly": [
+            "-Iunix/shim",
+        ],
+        "@io_bazel_rules_go//go/platform:freebsd": [
+            "-Iunix/shim",
+        ],
+        "@io_bazel_rules_go//go/platform:ios": [
+            "-Iunix/shim",
+        ],
+        "@io_bazel_rules_go//go/platform:linux": [
+            "-Iunix/src -Iunix/src/c-libedit -Iunix/src/c-libedit/editline -Iunix/src/c-libedit/linux-build -D_GNU_SOURCE",
+            "-Wno-unused-result -Wno-pointer-sign",
+        ],
+        "@io_bazel_rules_go//go/platform:netbsd": [
+            "-Iunix/shim",
+        ],
+        "@io_bazel_rules_go//go/platform:openbsd": [
+            "-Iunix/shim",
+        ],
+        "//conditions:default": [],
+    }),
+    importpath = "",
+    visibility = ["//visibility:public"],
+    deps = ["@com_github_knz_go_libedit//common:go_default_library"],
+)
+
+go_library(
+    name = "unix",
+    cdeps = [":libedit_unix_cdeps"], # TODO: make this work on non-mac.
```

Comment:
Can you explain what doesn't work

Reply:
it needs to do a `select` based on operating system, and compile `unix/src` instead for linux.
also very draft-y at the moment -- just building this atm to make it easier to replace as a go module.

Reply:
Thanks for explaining. Give me a heads up when you need something from me